libraries/AP_Math/tests/test_math.cpp

```diff
@@ -0,0 +1,309 @@
+#include <AP_gtest.h>
+#include <AP_Math/AP_Math.h>
+
+#define SQRT_2 1.4142135623730951f
+
+
+TEST(ZeroTest, zero)
+{
+  #define TEST_IS_ZERO(_test, _result) { \
+    bool bZero = is_zero(_test); \
+    EXPECT_EQ(bZero, _result) << "is_zero(): floating point is_zero comparator failure"; \
+  }
+  
+  TEST_IS_ZERO(0.1,      false);
+  TEST_IS_ZERO(0.0001,   false);
+  TEST_IS_ZERO(0.f,      true);
+  TEST_IS_ZERO(FLT_MIN,  true);
+  TEST_IS_ZERO(-FLT_MIN, true);
+}
+
+TEST(EqualTest, equal)
+{
+  #define TEST_IS_EQUAL(_test1, _test2, _result) { \
+    bool bEqual = is_equal(_test1, _test2); \
+    EXPECT_EQ(bEqual, _result) << "is_equal(): floating point is_equal comparator failure"; \
+  }
+  
+  TEST_IS_EQUAL(0.1,     0.10001,           false);
+  TEST_IS_EQUAL(0.1,     -0.1001,           false);
+  TEST_IS_EQUAL(0.f,     0.0f,              true);
```

Comment:
We should pass on non zero value and negative value too
TEST_IS_EQUAL(10.f,     10.f,              true);
TEST_IS_EQUAL(-10.f,     -10.f,              true);

And int 
TEST_IS_EQUAL(10,    10,              true);
TEST_IS_EQUAL(-10,   -10,              true);
TEST_IS_EQUAL(10,    11,              false);
TEST_IS_EQUAL(10,    -10,              false);